### PR TITLE
set pre-release to be semantic versions

### PR DESCRIPTION
**Pull request by yaronkaikov**
Today we are not using semantic versioning for our pre-release, Let's switch to the following convention for both rpm and deb packages: 
```
4.5.0~rc0 .. 4.5.0~rc1 .. 4.5.0
```

For that we need to remove the replacing of `.rc` with `~rc`

Closes: https://github.com/scylladb/scylla/issues/9543

### CODE REVIEW

**Comment by yaronkaikov** [comment]
Will open a relevant PR in machine-image and Scylla-pkg to align those changes

File: dist/debian/debian_files_gen.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ class DebianFilesTemplate(string.Template):
     product = f.read().strip()
 
 with open('build/SCYLLA-VERSION-FILE') as f:
-    version = f.read().strip().replace('.rc', '~rc').replace('_', '-')
+    version = f.read().strip()
```
Comment (avikivity) [inline dist/debian/debian_files_gen.py:32]: I suggest to drop this after we complete a cycle with the new names. It doesn't hurt if the names already have ~rc.

Also, why did you change the second .replace?

Comment (yaronkaikov) [reply dist/debian/debian_files_gen.py:32]: Ok, we can actually wait. As for the second part, it's not being used since our version syntax is `5.1.dev` or `5.1.0~rc` , so there is no `_` in version string (this is the original commit https://github.com/scylladb/scylla/commit/9de65f26ded98fc2570951b680aea302b96d3eac)
